src/compositor/hwcimage.cpp

```diff
@@ -587,6 +600,11 @@ void HwcImageTexture::bind()
     }
 }
 
+void hwcimage_delete_texture(void *, void *callbackData)
+{
+    delete (QObject *) callbackData;
```

Comment:
And if it's called from another thread, maybe even `static_cast<QObject *>(callbackData)->deleteLater();`?